ion-schema/src/system.rs

```diff
@@ -931,30 +1039,57 @@ impl Resolver {
                     _ => Err(error),
                 },
                 Ok(schema_content) => {
-                    let isl = self.isl_schema_from_elements(schema_content.into_iter(), id)?;
-                    self.schema_from_isl_schema(isl, id, type_store, load_isl_import)
+                    // ISL version marker regex
+                    let isl_version_marker = Regex::new(r"^\$ion_schema_\d.*$").unwrap();
+                    let mut isl_version = IonSchemaLanguageVersion::V1_0;
+
+                    // find the ISL version
+                    for value in &schema_content {
+                        // verify if value is an ISL version marker and if it has valid format
+                        if value.ion_type() == IonType::Symbol
+                            && isl_version_marker.is_match(value.as_str().unwrap())
+                        {
+                            // This implementation supports Ion Schema 1.0 and Ion Schema 2.0
+                            isl_version = match value.as_str().unwrap() {
+                                "$ion_schema_1_0" => IonSchemaLanguageVersion::V1_0,
+                                "$ion_schema_2_0" => IonSchemaLanguageVersion::V2_0,
```

Comment:
It should probably break out of the loop immediately after finding either of these.